llvm/lib/Target/Mips/MipsMSAInstrInfo.td

```diff
@@ -249,67 +249,67 @@ class SplatComplexPattern<Operand opclass, ValueType ty, int numops, string fn,
 }
 
 def vsplati8_uimm3 : SplatComplexPattern<vsplat_uimm3, v16i8, 1,
-                                         "selectVSplatUimm3",
+                                         "selectVSplatUimm<3>",
```

Comment:
We can further reduce duplicate code with foreach:
```
foreach I = {2, 4, 5, 8} in
ef vsplati8_uimm # I : SplatComplexPattern<vsplat_uimm # I, v16i8, 1,
                                          "selectVSplatUimm<"# I #">",
```
Up to you.

Reply:
I might do some other refactorings here. I'm not sure we need one of these for every element size.